### ISSUE TITLE
string comparison compiler warning in main.cc

### ISSUE THREAD

**Issue by agriggio** [artpixls/ART#120]
**[Original report](https://bitbucket.org/agriggio/art/issues/120) by Danny Heijl (Bitbucket: [dheijl](https://bitbucket.org/dheijl), GitHub: [dheijl](https://github.com/dheijl)).**

----------------------------------------

‌

```shell
/home/danny/programs/code-art/rtgui/main-cli.cc: In function ‘int main(int, char**)’:
/home/danny/programs/code-art/rtgui/main-cli.cc:148:36: warning: comparison with string literal results in unspecified behavior [-Waddress]
     } else if (DATA_SEARCH_PATH == ".") {
```

‌

**Comment by agriggio** [comment]
**Original comment by Danny Heijl (Bitbucket: [dheijl](https://bitbucket.org/dheijl), GitHub: [dheijl](https://github.com/dheijl)).**

----------------------------------------

If I change the line \(in [main.cc](http://main.cc) and [main-cli.cc](http://main-cli.cc)\) to

```c++
    } else if (std::string(DATA_SEARCH_PATH) == ".") {
```

the warning goes away and the comparison is OK.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

fixed wrong string comparisons

Fixes #120
